### ISSUE TITLE
PDF Generation is not working with AZ DevOps Pipeline

### ISSUE THREAD

**Issue by sourabhsomani** [dotnet/docfx#6313]
**Operating System**: `Windows`

**DocFX Version Used**: `docfx.console.2.56.1`

**Template used**: `pdf.default`

**Steps to Reproduce**:

# Functional impact
PDF Build fails in Azure DevOps.

**Steps to Reproduce:**
1. Get Sources from repository
2. Install **wkhtmltopdf** using **Chocolaty** using following command
`choco install wkhtmltopdf -y`
3. Generate pdf using **docfx.json**. We are just testing pdf generation and our pdf template is as follows.
```JSON
"pdf": {
    "name": "ais",
    "content": [
      {
        "files": [
          "index.md",
          "toc.yml"
        ]
      },
      {
        "files": [
          "Cloud Readiness/**.md",
          "Cloud Readiness/toc.yml",
          "Migrate/**.md",
          "Migrate/toc.yml",
          "Workload Landing Zone/**.md",
          "Workload Landing Zone/toc.yml",
          "Plan/**.md",
          "Plan/toc.yml",
          "Strategy/**.md",
          "Strategy/toc.yml",
          "Backlog/toc.yml",
          "Backlog/**.md"
        ],
        "src": ".."
      }
    ],
    "resource": [
      {
        "files": [
          "Cloud Readiness/**.png",
          "Cloud Readiness/**.jpg",
          "Migrate/**.png",
          "Migrate/**.jpg",
          "Workload Landing Zone/**.png",
          "Workload Landing Zone/**.jpg",
          "images/**.png",
          "Plan/**.png",
          "Plan/**.jpg",
          "Strategy/**.png",
          "Strategy/**.jpg",
          "Strategy/**.pptx"
        ],
        "src": ".."
      }
    ],
    
    "dest": "_site/static/files",
    "outline": "NoOutline",
    
    "noStdin": true,
    "keepRawFiles": true
  }

```
**Expected Behavior:**
Create PDF file

**Actual Behavior:**
PDF file is not create with these errors in the Build Pipeline.

```
2020-07-25T07:27:37.6981329Z [20-07-25 07:27:27.286]Error:[PdfCommand.PDF]Error happen when converting Backlog/toc.json to Pdf. Details: iTextSharp.text.exceptions.InvalidPdfException: PDF header signature not found.
2020-07-25T07:27:37.6982085Z    at iTextSharp.text.pdf.PdfReader..ctor(IRandomAccessSource byteSource, Boolean partialRead, Byte[] ownerPassword, X509Certificate certificate, ICipherParameters certificateKey, Boolean closeSourceOnConstructorError)
2020-07-25T07:27:37.6982599Z    at iTextSharp.text.pdf.PdfReader..ctor(Stream isp)
2020-07-25T07:27:37.6982899Z    at Microsoft.DocAsCode.HtmlToPdf.HtmlToPdfConverter.SaveCore(Stream stream)
2020-07-25T07:27:37.6983243Z    at Microsoft.DocAsCode.HtmlToPdf.HtmlToPdfConverter.Save(String outputFileName)
2020-07-25T07:27:37.6983662Z    at Microsoft.DocAsCode.HtmlToPdf.ConvertWrapper.<>c__DisplayClass7_0.<ConvertCore>b__1(ManifestItem tocFile)
2020-07-25T07:27:37.6984252Z [20-07-25 07:27:27.286]Error:[PdfCommand.PDF]Error happen when converting Plan/toc.json to Pdf. Details: iTextSharp.text.exceptions.InvalidPdfException: PDF header signature not found.
2020-07-25T07:27:37.6985013Z    at iTextSharp.text.pdf.PdfReader..ctor(IRandomAccessSource byteSource, Boolean partialRead, Byte[] ownerPassword, X509Certificate certificate, ICipherParameters certificateKey, Boolean closeSourceOnConstructorError)
2020-07-25T07:27:37.6985557Z    at iTextSharp.text.pdf.PdfReader..ctor(Stream isp)
2020-07-25T07:27:37.6985861Z    at Microsoft.DocAsCode.HtmlToPdf.HtmlToPdfConverter.SaveCore(Stream stream)
2020-07-25T07:27:37.6986214Z    at Microsoft.DocAsCode.HtmlToPdf.HtmlToPdfConverter.Save(String outputFileName)
2020-07-25T07:27:37.6986602Z    at Microsoft.DocAsCode.HtmlToPdf.ConvertWrapper.<>c__DisplayClass7_0.<ConvertCore>b__1(ManifestItem tocFile)
2020-07-25T07:27:37.6987271Z [20-07-25 07:27:30.905]Error:[PdfCommand.PDF]Error happen when converting Strategy/toc.json to Pdf. Details: iTextSharp.text.exceptions.InvalidPdfException: PDF header signature not found.
2020-07-25T07:27:37.6988044Z    at iTextSharp.text.pdf.PdfReader..ctor(IRandomAccessSource byteSource, Boolean partialRead, Byte[] ownerPassword, X509Certificate certificate, ICipherParameters certificateKey, Boolean closeSourceOnConstructorError)
2020-07-25T07:27:37.6988540Z    at iTextSharp.text.pdf.PdfReader..ctor(Stream isp)
2020-07-25T07:27:37.6988839Z    at Microsoft.DocAsCode.HtmlToPdf.HtmlToPdfConverter.SaveCore(Stream stream)
2020-07-25T07:27:37.6989192Z    at Microsoft.DocAsCode.HtmlToPdf.HtmlToPdfConverter.Save(String outputFileName)
2020-07-25T07:27:37.6989579Z    at Microsoft.DocAsCode.HtmlToPdf.ConvertWrapper.<>c__DisplayClass7_0.<ConvertCore>b__1(ManifestItem tocFile)
2020-07-25T07:27:37.6990187Z [20-07-25 07:27:30.919]Error:[PdfCommand.PDF]Error happen when converting Cloud Readiness/toc.json to Pdf. Details: iTextSharp.text.exceptions.InvalidPdfException: PDF header signature not found.
2020-07-25T07:27:37.6991020Z    at iTextSharp.text.pdf.PdfReader..ctor(IRandomAccessSource byteSource, Boolean partialRead, Byte[] ownerPassword, X509Certificate certificate, ICipherParameters certificateKey, Boolean closeSourceOnConstructorError)
2020-07-25T07:27:37.6991515Z    at iTextSharp.text.pdf.PdfReader..ctor(Stream isp)
2020-07-25T07:27:37.6991829Z    at Microsoft.DocAsCode.HtmlToPdf.HtmlToPdfConverter.SaveCore(Stream stream)
2020-07-25T07:27:37.6992168Z    at Microsoft.DocAsCode.HtmlToPdf.HtmlToPdfConverter.Save(String outputFileName)
2020-07-25T07:27:37.6992556Z    at Microsoft.DocAsCode.HtmlToPdf.ConvertWrapper.<>c__DisplayClass7_0.<ConvertCore>b__1(ManifestItem tocFile)
2020-07-25T07:27:37.6993189Z [20-07-25 07:27:34.507]Error:[PdfCommand.PDF]Error happen when converting Workload Landing Zone/toc.json to Pdf. Details: iTextSharp.text.exceptions.InvalidPdfException: PDF header signature not found.
2020-07-25T07:27:37.6994003Z    at iTextSharp.text.pdf.PdfReader..ctor(IRandomAccessSource byteSource, Boolean partialRead, Byte[] ownerPassword, X509Certificate certificate, ICipherParameters certificateKey, Boolean closeSourceOnConstructorError)
2020-07-25T07:27:37.6994516Z    at iTextSharp.text.pdf.PdfReader..ctor(Stream isp)
2020-07-25T07:27:37.6994811Z    at Microsoft.DocAsCode.HtmlToPdf.HtmlToPdfConverter.SaveCore(Stream stream)
2020-07-25T07:27:37.6995151Z    at Microsoft.DocAsCode.HtmlToPdf.HtmlToPdfConverter.Save(String outputFileName)
2020-07-25T07:27:37.6995554Z    at Microsoft.DocAsCode.HtmlToPdf.ConvertWrapper.<>c__DisplayClass7_0.<ConvertCore>b__1(ManifestItem tocFile)
2020-07-25T07:27:37.6996150Z [20-07-25 07:27:34.509]Error:[PdfCommand.PDF]Error happen when converting Migrate/toc.json to Pdf. Details: iTextSharp.text.exceptions.InvalidPdfException: PDF header signature not found.
2020-07-25T07:27:37.6997068Z    at iTextSharp.text.pdf.PdfReader..ctor(IRandomAccessSource byteSource, Boolean partialRead, Byte[] ownerPassword, X509Certificate certificate, ICipherParameters certificateKey, Boolean closeSourceOnConstructorError)
2020-07-25T07:27:37.6997620Z    at iTextSharp.text.pdf.PdfReader..ctor(Stream isp)
2020-07-25T07:27:37.6997921Z    at Microsoft.DocAsCode.HtmlToPdf.HtmlToPdfConverter.SaveCore(Stream stream)
2020-07-25T07:27:37.6998258Z    at Microsoft.DocAsCode.HtmlToPdf.HtmlToPdfConverter.Save(String outputFileName)
2020-07-25T07:27:37.6998660Z    at Microsoft.DocAsCode.HtmlToPdf.ConvertWrapper.<>c__DisplayClass7_0.<ConvertCore>b__1(ManifestItem tocFile)
2020-07-25T07:27:37.6999266Z [20-07-25 07:27:37.616]Error:[PdfCommand.PDF]Error happen when converting toc.json to Pdf. Details: iTextSharp.text.exceptions.InvalidPdfException: PDF header signature not found.
2020-07-25T07:27:37.7000368Z    at iTextSharp.text.pdf.PdfReader..ctor(IRandomAccessSource byteSource, Boolean partialRead, Byte[] ownerPassword, X509Certificate certificate, ICipherParameters certificateKey, Boolean closeSourceOnConstructorError)
2020-07-25T07:27:37.7000899Z    at iTextSharp.text.pdf.PdfReader..ctor(Stream isp)
2020-07-25T07:27:37.7001196Z    at Microsoft.DocAsCode.HtmlToPdf.HtmlToPdfConverter.SaveCore(Stream stream)
2020-07-25T07:27:37.7001557Z    at Microsoft.DocAsCode.HtmlToPdf.HtmlToPdfConverter.Save(String outputFileName)
2020-07-25T07:27:37.7001947Z    at Microsoft.DocAsCode.HtmlToPdf.ConvertWrapper.<>c__DisplayClass7_0.<ConvertCore>b__1(ManifestItem tocFile)
2020-07-25T07:27:37.7002484Z [20-07-25 07:27:37.619]Info:[PdfCommand.PDF]Completed Scope:PdfCommand.PDF in 15137.5244 milliseconds.
2020-07-25T07:27:37.7003571Z [20-07-25 07:27:37.619]Info:[PdfCommand]Completed Scope:PdfCommand in 19556.6278 milliseconds.
2020-07-25T07:27:37.7004211Z [20-07-25 07:27:37.619]Info:Completed in 26514.2262 milliseconds
2020-07-25T07:27:37.7004411Z 
2020-07-25T07:27:37.7004526Z 
2020-07-25T07:27:37.7004666Z Build failed.
2020-07-25T07:27:37.6981329Z [20-07-25 07:27:27.286]Error:[PdfCommand.PDF]Error happen when converting Backlog/toc.json to Pdf. Details: iTextSharp.text.exceptions.InvalidPdfException: PDF header signature not found.
2020-07-25T07:27:37.6982085Z    at iTextSharp.text.pdf.PdfReader..ctor(IRandomAccessSource byteSource, Boolean partialRead, Byte[] ownerPassword, X509Certificate certificate, ICipherParameters certificateKey, Boolean closeSourceOnConstructorError)
2020-07-25T07:27:37.6982599Z    at iTextSharp.text.pdf.PdfReader..ctor(Stream isp)
2020-07-25T07:27:37.6982899Z    at Microsoft.DocAsCode.HtmlToPdf.HtmlToPdfConverter.SaveCore(Stream stream)
2020-07-25T07:27:37.6983243Z    at Microsoft.DocAsCode.HtmlToPdf.HtmlToPdfConverter.Save(String outputFileName)
2020-07-25T07:27:37.6983662Z    at Microsoft.DocAsCode.HtmlToPdf.ConvertWrapper.<>c__DisplayClass7_0.<ConvertCore>b__1(ManifestItem tocFile)
2020-07-25T07:27:37.6984252Z [20-07-25 07:27:27.286]Error:[PdfCommand.PDF]Error happen when converting Plan/toc.json to Pdf. Details: iTextSharp.text.exceptions.InvalidPdfException: PDF header signature not found.
2020-07-25T07:27:37.6985013Z    at iTextSharp.text.pdf.PdfReader..ctor(IRandomAccessSource byteSource, Boolean partialRead, Byte[] ownerPassword, X509Certificate certificate, ICipherParameters certificateKey, Boolean closeSourceOnConstructorError)
2020-07-25T07:27:37.6985557Z    at iTextSharp.text.pdf.PdfReader..ctor(Stream isp)
2020-07-25T07:27:37.6985861Z    at Microsoft.DocAsCode.HtmlToPdf.HtmlToPdfConverter.SaveCore(Stream stream)
2020-07-25T07:27:37.6986214Z    at Microsoft.DocAsCode.HtmlToPdf.HtmlToPdfConverter.Save(String outputFileName)
2020-07-25T07:27:37.6986602Z    at Microsoft.DocAsCode.HtmlToPdf.ConvertWrapper.<>c__DisplayClass7_0.<ConvertCore>b__1(ManifestItem tocFile)
2020-07-25T07:27:37.6987271Z [20-07-25 07:27:30.905]Error:[PdfCommand.PDF]Error happen when converting Strategy/toc.json to Pdf. Details: iTextSharp.text.exceptions.InvalidPdfException: PDF header signature not found.
2020-07-25T07:27:37.6988044Z    at iTextSharp.text.pdf.PdfReader..ctor(IRandomAccessSource byteSource, Boolean partialRead, Byte[] ownerPassword, X509Certificate certificate, ICipherParameters certificateKey, Boolean closeSourceOnConstructorError)
2020-07-25T07:27:37.6988540Z    at iTextSharp.text.pdf.PdfReader..ctor(Stream isp)
2020-07-25T07:27:37.6988839Z    at Microsoft.DocAsCode.HtmlToPdf.HtmlToPdfConverter.SaveCore(Stream stream)
2020-07-25T07:27:37.6989192Z    at Microsoft.DocAsCode.HtmlToPdf.HtmlToPdfConverter.Save(String outputFileName)
2020-07-25T07:27:37.6989579Z    at Microsoft.DocAsCode.HtmlToPdf.ConvertWrapper.<>c__DisplayClass7_0.<ConvertCore>b__1(ManifestItem tocFile)
2020-07-25T07:27:37.6990187Z [20-07-25 07:27:30.919]Error:[PdfCommand.PDF]Error happen when converting Cloud Readiness/toc.json to Pdf. Details: iTextSharp.text.exceptions.InvalidPdfException: PDF header signature not found.
2020-07-25T07:27:37.6991020Z    at iTextSharp.text.pdf.PdfReader..ctor(IRandomAccessSource byteSource, Boolean partialRead, Byte[] ownerPassword, X509Certificate certificate, ICipherParameters certificateKey, Boolean closeSourceOnConstructorError)
2020-07-25T07:27:37.6991515Z    at iTextSharp.text.pdf.PdfReader..ctor(Stream isp)
2020-07-25T07:27:37.6991829Z    at Microsoft.DocAsCode.HtmlToPdf.HtmlToPdfConverter.SaveCore(Stream stream)
2020-07-25T07:27:37.6992168Z    at Microsoft.DocAsCode.HtmlToPdf.HtmlToPdfConverter.Save(String outputFileName)
2020-07-25T07:27:37.6992556Z    at Microsoft.DocAsCode.HtmlToPdf.ConvertWrapper.<>c__DisplayClass7_0.<ConvertCore>b__1(ManifestItem tocFile)
2020-07-25T07:27:37.6993189Z [20-07-25 07:27:34.507]Error:[PdfCommand.PDF]Error happen when converting Workload Landing Zone/toc.json to Pdf. Details: iTextSharp.text.exceptions.InvalidPdfException: PDF header signature not found.
2020-07-25T07:27:37.6994003Z    at iTextSharp.text.pdf.PdfReader..ctor(IRandomAccessSource byteSource, Boolean partialRead, Byte[] ownerPassword, X509Certificate certificate, ICipherParameters certificateKey, Boolean closeSourceOnConstructorError)
2020-07-25T07:27:37.6994516Z    at iTextSharp.text.pdf.PdfReader..ctor(Stream isp)
2020-07-25T07:27:37.6994811Z    at Microsoft.DocAsCode.HtmlToPdf.HtmlToPdfConverter.SaveCore(Stream stream)
2020-07-25T07:27:37.6995151Z    at Microsoft.DocAsCode.HtmlToPdf.HtmlToPdfConverter.Save(String outputFileName)
2020-07-25T07:27:37.6995554Z    at Microsoft.DocAsCode.HtmlToPdf.ConvertWrapper.<>c__DisplayClass7_0.<ConvertCore>b__1(ManifestItem tocFile)
2020-07-25T07:27:37.6996150Z [20-07-25 07:27:34.509]Error:[PdfCommand.PDF]Error happen when converting Migrate/toc.json to Pdf. Details: iTextSharp.text.exceptions.InvalidPdfException: PDF header signature not found.
2020-07-25T07:27:37.6997068Z    at iTextSharp.text.pdf.PdfReader..ctor(IRandomAccessSource byteSource, Boolean partialRead, Byte[] ownerPassword, X509Certificate certificate, ICipherParameters certificateKey, Boolean closeSourceOnConstructorError)
2020-07-25T07:27:37.6997620Z    at iTextSharp.text.pdf.PdfReader..ctor(Stream isp)
2020-07-25T07:27:37.6997921Z    at Microsoft.DocAsCode.HtmlToPdf.HtmlToPdfConverter.SaveCore(Stream stream)
2020-07-25T07:27:37.6998258Z    at Microsoft.DocAsCode.HtmlToPdf.HtmlToPdfConverter.Save(String outputFileName)
2020-07-25T07:27:37.6998660Z    at Microsoft.DocAsCode.HtmlToPdf.ConvertWrapper.<>c__DisplayClass7_0.<ConvertCore>b__1(ManifestItem tocFile)
2020-07-25T07:27:37.6999266Z [20-07-25 07:27:37.616]Error:[PdfCommand.PDF]Error happen when converting toc.json to Pdf. Details: iTextSharp.text.exceptions.InvalidPdfException: PDF header signature not found.
2020-07-25T07:27:37.7000368Z    at iTextSharp.text.pdf.PdfReader..ctor(IRandomAccessSource byteSource, Boolean partialRead, Byte[] ownerPassword, X509Certificate certificate, ICipherParameters certificateKey, Boolean closeSourceOnConstructorError)
2020-07-25T07:27:37.7000899Z    at iTextSharp.text.pdf.PdfReader..ctor(Stream isp)
2020-07-25T07:27:37.7001196Z    at Microsoft.DocAsCode.HtmlToPdf.HtmlToPdfConverter.SaveCore(Stream stream)
2020-07-25T07:27:37.7001557Z    at Microsoft.DocAsCode.HtmlToPdf.HtmlToPdfConverter.Save(String outputFileName)
2020-07-25T07:27:37.7001947Z    at Microsoft.DocAsCode.HtmlToPdf.ConvertWrapper.<>c__DisplayClass7_0.<ConvertCore>b__1(ManifestItem tocFile)
2020-07-25T07:27:37.7002484Z [20-07-25 07:27:37.619]Info:[PdfCommand.PDF]Completed Scope:PdfCommand.PDF in 15137.5244 milliseconds.
2020-07-25T07:27:37.7003571Z [20-07-25 07:27:37.619]Info:[PdfCommand]Completed Scope:PdfCommand in 19556.6278 milliseconds.
2020-07-25T07:27:37.7004211Z [20-07-25 07:27:37.619]Info:Completed in 26514.2262 milliseconds
2020-07-25T07:27:37.7004411Z 
2020-07-25T07:27:37.7004526Z 
2020-07-25T07:27:37.7004666Z Build failed.
```

**Comment by icnocop** [comment]
Hi @sourabhsomani.

Can you run docfx with the `--logLevel Verbose` command line option and see if there are any relevant messages?

Can you share a repo where this can be reproduced?

Thank you.

**Comment by yufeih** [comment]
Addressed in v2.73.0 with [a new PDF engine](https://dotnet.github.io/docfx/docs/pdf.html).
